packages/ts/form/src/Models.ts

```diff
@@ -34,57 +21,38 @@ export function hasFromString<T>(model: AbstractModel<T>): model is AbstractMode
   return _fromString in model;
 }
 
-export interface HasValue<T> {
-  value?: T;
-}
+export type Value<M> = M extends AbstractModel<infer T> ? T : never;
+
+export const _modelDetachedParent = { $value: undefined };
+
+export type ModelParent = AbstractModel | BinderNode | typeof _modelDetachedParent;
 
-export type ModelParent<T> = AbstractModel<any> | HasValue<T>;
-export type ModelValue<M extends AbstractModel<unknown>> = ReturnType<M['valueOf']>;
+export type EmptyModelConstructor<M> = new (
+  parent: typeof _modelDetachedParent,
+  key: '$value$',
+  optional: boolean,
+) => M;
 
-export interface ModelConstructor<T, M extends AbstractModel<T>> {
-  new (parent: ModelParent<T>, key: keyof any, optional: boolean, ...args: any[]): M;
-  createEmptyValue(): T;
+export function _createDetachedModel<M extends AbstractModel>(type: EmptyModelConstructor<M>): M {
```

Comment:
Let's not use underscore for anything except Symbols.
```suggestion
export function createDetachedModel<M extends AbstractModel>(type: EmptyModelConstructor<M>): M {
```